src/amp-url-creator.js

```diff
@@ -58,25 +59,30 @@ const MAX_DOMAIN_LABEL_LENGTH_ = 63;
  * @param {string} url The complete publisher url.
  * @param {object} initParams Params containing origin, etc.
  * @param {string} opt_cacheUrlAuthority
- * @param {string} opt_viewer_js_version
- * @return {string} the Cache Url.
+ * @param {string} opt_viewerJsVersion
+ * @return {!Promise}
```

Comment:
The sha256 returns a Promise which makes affects all the other method implementations